OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/IamFetchReadyCondition.kt

```diff
@@ -0,0 +1,52 @@
+package com.onesignal.inAppMessages.internal
+
+import com.onesignal.common.consistency.enums.IamFetchRywTokenKey
+import com.onesignal.common.consistency.impl.ConsistencyManager
+import com.onesignal.common.consistency.models.ICondition
+import com.onesignal.common.consistency.models.IConsistencyKeyEnum
+
+/**
+ * Used for read your write consistency when fetching In-App Messages.
+ *
+ * Params:
+ *  id : String - the index of the RYW token map
+ */
+class IamFetchReadyCondition(
+    private val id: String,
+) : ICondition {
+    override fun isMet(indexedTokens: Map<String, Map<IConsistencyKeyEnum, String?>>): Boolean {
+        val tokenMap = indexedTokens[id] ?: return false
+        val userUpdateTokenSet = tokenMap[IamFetchRywTokenKey.USER_UPDATE] != null
+        val subscriptionUpdateTokenSet = tokenMap[IamFetchRywTokenKey.SUBSCRIPTION_UPDATE] != null
+
+        /**
+         * If the RYW token was not included in one of the 4 endpoints (PATCH & POST * user,
+         * subscription), we should resolve the condition so as to not block IAM fetch unnecessarily
+         * This can happen if there is a regression where we stop including the tokens in the res
+         * body or we turn off the feature temporarily
+         */
+        val subscriptionOrUserTokenWasNotSet =
+            tokenMap[IamFetchRywTokenKey.SUBSCRIPTION_UPDATE] == ConsistencyManager.RYW_TOKEN_NOT_PROVIDED ||
```

Comment:
if the value of `RYW_TOKEN_NOT_PROVIDED` is "0" but actual ryw-tokens are always 20 characters long won't this never be true?

Reply:
It's a magic number. If the value is null when set, [we set it to this constant](https://github.com/OneSignal/OneSignal-Android-SDK/pull/2168/files#diff-ebc9431e7f9803eb553d3c75bbbbc540596a279fe76e419255b4fdcf0ade749aR48). We then use it to [exclude the value from the IAM fetch](https://github.com/OneSignal/OneSignal-Android-SDK/pull/2168/files#diff-3f270602e75b4896d8f56cfac018353aa0b7f85e72e5c7ea25423202d7ca9ed9R217).

The alternative is to introduce a new ConsistencyManager method to force-resolve some or all conditions (i.e. blocking IAM fetch)

Reply:
Update: I re-architected it to avoid the magic number. We now call [`resolveConditionsWithId` to unblock the execution](https://github.com/OneSignal/OneSignal-Android-SDK/pull/2168/commits/51aaa8b139ad6a9048fb99f9890f44b094f62792)